src/VisualStudio/Core/Def/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs

```diff
@@ -136,11 +136,11 @@ internal Metadata GetMetadata(string fullPath, DateTime snapshotTimestamp)
             }
 
             // use temporary storage
-            var storages = new List<ITemporaryStreamStorage>();
+            var storages = new List<TemporaryStorageServiceFactory.TemporaryStorageService.TemporaryStreamStorage>();
```

Comment:
>TemporaryStorageServiceFactory.TemporaryStorageService.TemporaryStreamStorage [](http://example.com/codeflow?start=36&length=77)

Oh, dear.... triple nesting!

Reply:
"clunky" :-D

Reply:
Yeah, that's why I generally prefer nested factory pattern.

Reply:
Wfm.  I'll move to that ASAP